internal/stream/stream_test.go

```diff
@@ -250,3 +251,150 @@ func TestPushStream_Stop(t *testing.T) {
 	err := ps.Stop()
 	assert.Nil(t, err)
 }
+
+func TestNewPushStream_Failure(t *testing.T) {
+	ctx := context.Background()
+	ctrl := gomock.NewController(t)
+	subscriptionCoreMock := mocks1.NewMockICore(ctrl)
+	subscriberCoreMock := mocks2.NewMockICore(ctrl)
+	workerID := uuid.New().String()
+	httpConfig := &httpclient.Config{}
+	type args struct {
+		ctx              context.Context
+		nodeID           string
+		subName          string
+		subscriptionCore subscription.ICore
+		subscriberCore   subscriber.ICore
+		config           *httpclient.Config
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *PushStream
+		wantErr bool
+	}{
+		{
+			name: "New Push Stream Failure",
+			args: args{
+				ctx:              ctx,
+				nodeID:           workerID,
+				subName:          subName,
+				subscriptionCore: subscriptionCoreMock,
+				subscriberCore:   subscriberCoreMock,
+				config:           httpConfig,
+			},
+			want:    nil,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
```

Comment:
if its a single test case, no need to create a test struct array and then loop over tests.

Reply:
remove test structs